### PR TITLE
Refactor use of models and views, and add subregion fly-to transitions

### PR DESCRIPTION
**Pull request by caseycesari**
## Overview

Refactors the existing code to make better use of Backbone models and views, and to conform to Backbone best practices. These changes reduce the need for custom state management and rendering code, and take advantage of some built-in Backbone features.

This was all in anticipation of implementing fly-to transitions and detail views for subregions, which required coordination between the map and sidebar. This is made easier by instantiating the regions and subregions as models, which have state that can be accessed throughout the app.

When a user clicks on a subregion in the sidebar, the description of the subregion is animated in below the subregion button. Also, the map does a fly-to transition to the subregion location and opens the subregion popup. Selecting another subregion in the same region closes the initial popup and transitions to the selected region. Exiting the region closes the popup, removes the markers, and transitions back to the initial map state.

Connects #32 

## Demo

tnc1

## Note

The animations are smoother than they appear in the above gif.

## Testing Instructions

- Select a region, and then select various subregions in that region.
- Verify that when you select a subregion, the map focuses on that subregion and opens the subregion popup.
- Verify that exiting out of the region closes any open popups and removes the markers, like before.

### CODE REVIEW

**Comment by ajrobbins** [comment]
Where is the Open Map button in the popup? Is it just below the text/scrolled out of view?

**Comment by caseycesari** [comment]
> Where is the Open Map button in the popup? Is it just below the text/scrolled out of view?

Yeah. I set a max height on the popups because there are large differences between the amount of text for each subregion. It helps with the transitions if the popups are the similar/same height.

**Comment by kellyi** [comment]
Taking a look at this!

**Comment by kellyi** [comment]
This is working as described above in the testing instructions. Question: when I click to close an open popup, should that also have the effect of de-selecting a region?

File: src/config.json [hunk 1]
```diff
@@ -43,28 +45,32 @@
...
+                    "mapZoom": 6
                 },
                 {
                     "title": "U.S Virgin Islands",
                     "description": "In the US Virgin Islands, high resolution satellite imagery and GIS-based models are being used to identify and prioritize high risk and vulnerable coastal and marine sites subject to the effects of sea level rise, increasing storm surge and intensity, and altered precipitation patterns. The impacts of future SLR and storm surge scenarios are being validated by incorporating community perception, knowledge and historic events as a baseline to better understand how ecosystem-based adaptation solutions can help increase people and nature’s resilience to these impacts.",
                     "image": "assets/images/locations/virgin-islands.svg",
                     "mapUrl": "http://maps.coastalresilience.org/usvi/",
-                    "mapCenter": [17.72, -64.78]
+                    "mapCenter": [17.72, -64.78],
+                    "mapZoom": 6
```
Comment (kellyi) [inline src/config.json:73]: Once these changes are in we should update the configuration docs in the wiki.

File: src/index.html [hunk 2]
```diff
@@ -15,7 +15,7 @@
     <meta content="#ffffff" name="theme-color">
     <link href="assets/images/favicon.ico?" rel="shortcut icon" type="image/x-icon">
     <title>
-        Coastal Resilience – Mapping portal
+        Coastal Resilience – Mapping portal
```
Comment (kellyi) [inline src/index.html:18]: Possibly trivial but I can't tell what changed in this line. Different whitespace char?

File: src/js/views.js [hunk 3]
```diff
@@ -170,31 +207,60 @@ window.RegionDetailsView = Backbone.View.extend({
...
     render: function() {
         var self = this;
-
         this.$el.html(this.template(this.model.attributes));
 
         this.$el.find('#subregion-list').html(this.model.get('subregions').map(function(subregion) {
-            return new window.SubregionDetailsView({ model: subregion }).render().el;
+            var view = new window.SubregionDetailsView({ model: subregion });
+
+            self.listenTo(view, 'toggle', self.toggleSubregionDetails);
```
Comment (kellyi) [inline src/js/views.js]: Does this listener go out of scope when the view's destroyed?

File: src/config.json (hunk 1 above)
Comment (caseycesari) [reply src/config.json:73]: Good call.

File: src/index.html (hunk 2 above)
Comment (caseycesari) [reply src/index.html:18]: Yeah, just some whitespace.

**Comment by caseycesari** [comment]
>  when I click to close an open popup, should that also have the effect of de-selecting a region?

Good question. I'm not sure. Perhaps we can wait to adjust this once we get some feedback from TNC.

**Review by kellyi — APPROVED** [review]
This looks good to me! I tested it out in FF, Safari, Chrome, and IE11 and it's the new features & prior features are working correctly in each.

Made a few comments on the code but they're mostly just questions to clarify things. Nice work refactoring this -- I think it'll make it easier to work with moving forward.

File: src/js/views.js (hunk 3 above)
Comment (caseycesari) [reply src/js/views.js]: I read through some things (including this: https://github.com/jashkenas/backbone/issues/828), and I think we are in the clear. The listeners appear to get garbage collected when the DOM elements are removed. But good catch.

**Comment by caseycesari** [comment]
Thanks for the review!

**Pull request merged**